core/server/apps/amp/lib/router.js

```diff
@@ -42,21 +43,42 @@ function getPostData(req, res, next) {
             next();
         })
         .catch(function (err) {
-            if (err instanceof errors.NotFoundError) {
-                return next(err);
-            }
-
             next(err);
         });
 }
 
+function checkIfAMPIsEnabled(req, res, next) {
+    var ampIsEnabled = config.theme.amp;
+
+    if (ampIsEnabled) {
+        return next();
+    }
+
+    // CASE: we don't support amp pages for static pages
+    if (req.body.post && req.body.post.page) {
+        return errors.error404(req, res, next);
+    }
+
+    /**
+     * CASE: amp is disabled, we redirect the user to the requested post
```

Comment:
@AileenCGN: AMP discoverability is provided by the relevant link in the HTML page. When AMP is enabled we should a tag like this `<link rel="amphtml" href="https://myblog.com/mypost/amp/">` which is enough to tell Google to look even if they previously saw a 404 for that URL. Likewise when AMP is disabled we shouldn't be outputting any links to AMP pages.